lib/pdk/config.rb

```diff
@@ -0,0 +1,94 @@
+require 'pdk/config/errors'
+require 'pdk/config/json'
+require 'pdk/config/namespace'
+require 'pdk/config/validator'
+require 'pdk/config/value'
+require 'pdk/config/yaml'
+
+module PDK
+  def self.config
+    @config ||= PDK::Config.new
+  end
+
+  class Config
+    def user
+      @user ||= PDK::Config::JSON.new('user', file: PDK::Config.user_config_path) do
+        mount :module_defaults, PDK::Config::JSON.new(file: PDK.answers.answer_file_path)
+
+        mount :analytics, PDK::Config::YAML.new(file: PDK::Config.analytics_config_path) do
+          value :disabled do
+            validate PDK::Config::Validator.boolean
+            default_to { PDK::Config.bolt_analytics_config.fetch('disabled', true) }
+          end
+
+          value 'user-id' do
+            validate PDK::Config::Validator.uuid
+            default_to do
+              require 'securerandom'
+
+              PDK::Config.bolt_analytics_config.fetch('user-id', SecureRandom.uuid)
+            end
+          end
+        end
+      end
+    end
+
+    def self.bolt_analytics_config
+      PDK::Config::YAML.new(file: File.expand_path('~/.puppetlabs/bolt/analytics.yaml'))
+    end
+
+    def self.analytics_config_path
+      ENV['PDK_ANALYTICS_CONFIG'] || File.join(File.dirname(PDK::Util.configdir), 'puppetlabs', 'analytics.yml')
+    end
+
+    def self.user_config_path
+      File.join(PDK::Util.configdir, 'user_config.json')
+    end
+
+    def self.analytics_config_exist?
+      PDK::Util::Filesystem.file?(analytics_config_path)
+    end
+
+    def self.analytics_config_interview!
+      return unless PDK::CLI::Util.interactive?
+
+      pre_message = _(
+        'PDK collects anonymous usage information to help us understand how ' \
+        'it is being used and make decisions on how to improve it. You can ' \
+        'find out more about what data we collect and how it is used in the ' \
+        'PDK documentation at %{url}.',
+      ) % { url: 'https://puppet.com/docs/pdk/latest/pdk_install.html' }
+      post_message = _(
+        'You can opt in or out of the usage data collection at any time by ' \
+        'editing the analytics configuration file at %{path} and changing ' \
+        "the '%{key}' value.",
```

Comment:
We should maybe pre-word wrap these at 80 characters or something. Right now the terminal will wrap in the middle of a word. :)

Reply:
Also an extra newline after the "post_message" would help it to not get lost in the results of the actual command they ran.

Reply:
I've updated the logger formatter so that we can optionally word wrap the output which gives us the following:
```
pdk (INFO): PDK collects anonymous usage information to help us understand how
            it is being used and make decisions on how to improve it. You can
            find out more about what data we collect and how it is used in the
            PDK documentation at
            https://puppet.com/docs/pdk/latest/pdk_install.html.

[Q 1/1] Do you consent to the collection of anonymous PDK usage information?
--> Yes

pdk (INFO): You can opt in or out of the usage data collection at any time by
            editing the analytics configuration file at
            /home/tsharpe/.config/puppet/analytics.yml and changing the
            'disabled' value.
```